src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPatternsManager.java

```diff
@@ -1,47 +1,53 @@
 package edu.agh.wfiis.designpatterns.patternsUsageExample;
 
+import java.util.*;
+import java.io.*;
 
-import java.io.File;
-
-public class DesignPatternsManager {
-
-    public DesignPattern create(PatternName patternName) {
-        switch (patternName) {
-            case BUILDER: {
-                DesignPattern pattern = new DesignPattern("1","builder","builder pattern",null, true, false, false);
-                return pattern;
-            }
-            case DECORATOR: {
-                DesignPattern pattern = new DesignPattern("2","decorator","decorator pattern",null, true, false, false);
-                return pattern;
-            }
-            case STRATEGY: {
-                DesignPattern pattern = new DesignPattern("3","strategy","strategy pattern",null, false, false, true);
-                return pattern;
-            }
-            default: {
-                DesignPattern pattern = new UndefinedPattern();
-                return pattern;
-            }
+public class DesignPatternsManager implements Serializable {
+
+
+    private static volatile DesignPatternsManager instance;
+    private Map<String, DesignPattern> patterns = new HashMap<String, DesignPattern>();
+
+    public DesignPattern create(String patternName) {
```

Comment:
Może lepszą nazwą byłoby getDesignPattern albo coś podobnego? Create jest trochę mylące, bo metoda wyciąga wzorzec z mapy, a nie tworzy nowy

Reply:
To prawda, ale zewnętrzny klient będzie to odbierać identycznie, ponieważ nie może zmienić stanu obiektu `DesignPattern`, nie zauważy, że "tworząc" dostaje ciągle ten sam obiekt. Też, skoro nie może go zmienić, to nie będzie przypadku użycia, gdzie zajdzie potrzeba porównania: "czy np. `builderObject1` i `builderObject2` to ten sam obiekt?".

Reply:
To może być też mylące dla innego programisty.

Reply:
As long as the name of the class is DesignPatternsManager, this name is irrelevant